generated/serialization.h

```diff
@@ -0,0 +1,80 @@
+#ifndef DTOGEN_SERIALIZATION_H
+#define DTOGEN_SERIALIZATION_H
+
+#include <memory>
+#include <optional>
+
+#include "nlohmann/json.hpp"
+
+// TODO: move elsewhere
```

Comment:
and TODO: leave comment explaining why this is needed. (I.e. that some compilers are not smart enough -- or too pedantic -- when parsing `switch() {}` statements with enums).